### ISSUE TITLE
Rebus vs Rebus2 timeoutmanager

### ISSUE THREAD

**Issue by torangel** [rebus-org/Rebus#312]
Not sure if this should be discussed as a "issue" here at GitHub or at StackOwerflow. Trying here first :)

Using Rebus2 I am testing the two "timeoutmanagermodes" described in these blogposts:
http://mookid.dk/oncode/archives/3428
http://mookid.dk/oncode/archives/3421

Using Rebus2 with an internal timeoutservice is working perfectly. Timeouts are stored in SQL-server and are picked up and forwareded when they should. My usecase however is using Rebus in an Azure loadbalanced environment where using one single timeoutmanager is recommended?? (If each (possible loadbalanced)Rebus-endpoint should handle its own deferral this might introduce problems??-multiple processess reading/writing to SQL etc). As far as I can understand each Rebus instance should then send deferred messages to the single timeoutmanager/endpoint which takes care of storing and forwarding when due.

As far as I can see there are some changes between how Rebus and Rebus2 is handling deferral of messages.

Rebus2:
	https://github.com/rebus-org/Rebus/blob/1891aaa1852e6cf090a9af6afe2a15e4620a4d94/Rebus/Bus/RebusBus.cs#L137
	https://github.com/rebus-org/Rebus/blob/1891aaa1852e6cf090a9af6afe2a15e4620a4d94/Rebus/Bus/RebusBus.cs#L213

Rebus:
RebusBus.cs:
```
		if (storeTimeouts == null)
            {
                var timeoutManagerEndpointAddress = RebusConfigurationSection
                    .GetConfigurationValueOrDefault(s => s.TimeoutManagerAddress, "rebus.timeout");

                log.Info("Using external timeout manager with input queue '{0}'", timeoutManagerEndpointAddress);
                timeoutManagerAddress = timeoutManagerEndpointAddress;
            }
            else
            {
                log.Info("Using internal timeout manager");
                timeoutManagerAddress = this.receiveMessages.InputQueueAddress;
                dueTimeoutScheduler = new DueTimeoutScheduler(storeTimeouts, new DeferredMessageReDispatcher(this));
            }
```

To sum it up:
-Should I use a single and dedicated timeoutmanager given my loadbalanced scenario?
-If yes:Where can I specify the timeoutmanageraddress in Rebus2? -
--Looking at the code below it seems as Rebus2 always will use its own transportaddress as the address of the timeoutmanager?
```
var address = _transport.Address;
            if (string.IsNullOrWhiteSpace(address))
            {
                throw new InvalidOperationException("Cannot use ourselves as timeout manager because we're a one-way client");
            }
```

**Comment by mookid8000** [comment]
At the moment there's no way to configure the timeout manager address, as you've correctly discovered. This will change in the future when #313 is done :)

Rebus2's built-in timeout manager is perfectly capable of competing over the timeout storage though, so it should work well with the internal timeout manager, even when there's multiple instances of the same endpoint.

**Comment by torangel** [comment]
:+1:
